pkg/resource/deploy/step.go

```diff
@@ -385,11 +386,20 @@ type deleteProtectedError struct {
 }
 
 func (d deleteProtectedError) Error() string {
-	return fmt.Sprintf("resource %[1]q cannot be deleted\n"+
+	return d.err(runtime.GOOS)
+}
+
+func (d deleteProtectedError) err(os string) string {
+	msg := fmt.Sprintf("resource %[1]q cannot be deleted\n"+
 		"because it is protected. To unprotect the resource, "+
 		"either remove the `protect` flag from the resource in your Pulumi "+
 		"program and run `pulumi up`, or use the command:\n"+
 		"`pulumi state unprotect '%[1]s'`", d.urn)
+	if os == "windows" {
+		// Windows uses double-quotes instead of single-quotes.
+		msg = strings.ReplaceAll(msg, `'`, `"`)
```

Comment:
If urns can include `single-quotes`, [surprised-pikachu.jpg](https://example.com)

Reply:
They're not supposed to, but I wouldn't be surprised if the the system allowed it.